wikimedia/wikipedia-ios
# [YiR] Attach donate logic to YiR donate button

**Phabricator:** https://phabricator.wikimedia.org/T376046

### Notes
* This PR attaches the donate button action on the YiR slides to the donate coordinator
* It also adds a donate source item on the coordinator, but the implementation is not done here, it'll be handled in another task

### Test Steps
1. Set your device to a valid region (Italy or France)
2. Fresh install the app and login
3. Set the language to Italian or French
4. Read articles and generate 6+ article views in the History tab.
5. Go to developer settings and toggle on the Year in Review feature flag.
6. Go through the year in the review slides
7. Tap the donate button on the slides
8. A spinner should show when the config is loading 
9. The action sheet should show and the donation flow work as usual
